f/connectors/auditor2/auditor2.py

```diff
@@ -0,0 +1,234 @@
+        If any of the required Auditor 2 tables already exist.
+    """
+    logger.debug(
+        f"Checking if Auditor 2 project name '{project_name}' already exists in the database."
+    )
+
+    required_tables = [
+        f"auditor2_{project_name}_deployments",
+        f"auditor2_{project_name}_human_readable_labels",
+        f"auditor2_{project_name}_labels",
+        f"auditor2_{project_name}_sites",
+        f"auditor2_{project_name}_sound_file_summary",
+    ]
+
+    for table in required_tables:
+        if check_if_table_exists(conninfo(db), table):
+            raise ValueError(f"Auditor2 project name already in usage in '{table}'.")
+
+
+def extract_auditor2_archive(
+    auditor2_zip_path: Path,
+    storage_path: Path,
+) -> None:
+    """
+    Extracts a Auditor 2 ZIP archive to `storage_path`.
+
+    Parameters
+    ----------
+    auditor2_zip_path : Path
+        The path to the Auditor 2 ZIP file.
+    storage_path : str
+        The path to the root directory where the ZIP file will be extracted.
+    """
+    logger.debug(
+        f"Extracting Auditor 2 archive from {auditor2_zip_path} to {storage_path}"
+    )
+    if not auditor2_zip_path.exists():
+        raise FileNotFoundError(f"Auditor 2 ZIP file not found: {auditor2_zip_path}")
+    try:
+        shutil.unpack_archive(auditor2_zip_path, storage_path)
+        logger.info(f"Extracted Auditor 2 archive to: {storage_path}")
+    except shutil.ReadError as e:
+        raise ValueError(f"Unable to extract archive: {e}")
+
+
+def read_auditor2_csvs(storage_path: Path) -> dict[str, list[dict[str, str]]]:
+    """
+    Reads specific Auditor 2 CSVs from the extracted files and returns them as a dictionary.
+
+    The extracted directory should contain 5 CSV files, each with names including these substrings:
+    - deployments
+    - human_readable_labels
+    - labels
+    - sites
+    - sound_file_summary
+
+    Parameters
+    ----------
+    storage_path : Path
+        The path to the directory where the Auditor 2 files were extracted.
+
+    Returns
+    -------
+    dict[str, list[dict[str, str]]]
+        A dictionary where keys are the CSV file identifiers and values are lists of dictionaries containing the CSV data.
+    """
+    required_keys = [
+        "deployments",
+        "human_readable_labels",
+        "labels",
+        "sites",
+        "sound_file_summary",
+    ]
+
+    logger.debug(
+        f"Reading Auditor 2 CSV files from: {storage_path}, looking for keys: {required_keys}"
+    )
+
+    csv_files = list(storage_path.glob("*.csv"))
+    auditor2_tables = {}
+    found_keys = {}
+
+    for file in csv_files:
+        for key in required_keys:
+            if key in file.stem:
+                with file.open("r", encoding="utf-8") as f:
+                    reader = csv.DictReader(f)
+                    rows = list(reader)
+                auditor2_tables[key] = rows
+                found_keys[key] = True
+                break
```

Comment:
I notice these are substrings, so could produce subtle bugs when you grab one.
For example, say the ZIP contains `foo_sites_20250505.csv` and `bar_sites_20250505.csv`.

Consider adding a check that each required_key is found exactly once.  One idea, you could use a `from collections import Counter` for `found_keys` instead of `dict`, and then later assert all values == 1.  Or inside these `for` loops you can just check if `found_keys` is already set, but in this case please make it a `set()` instead of dict.